### PR TITLE
add StoreReclaims::Ignore

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

**Comment by jeffwashington** [comment]
@HaoranYi this allows more upsert calls to not care about returning reclaims. Shrink is one example.

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -7697,6 +7718,7 @@ impl AccountsDb {
         is_cached_store: bool,
```
Comment (HaoranYi) [inline runtime/src/accounts_db.rs:7718]: Is `is_cached_store` still needed? Does reclaim enum tell us `is_cached_store`?

**Comment by jeffwashington** [comment]
> Now we don't reclaim for shrinked accounts? Will that take more space for shrinked accounts?

reclaim is how we update an append vec to notify it that it has less useful accounts in it. This causes the append vec's 'count' to decrease. When it get to 0, it is dead and maybe the whole slot is dead.
We also identify candidates for shrinking in this function.
In the case I'm attacking, we are already shrinking the append vec. We know that all accounts in the append vec will either be updated to be in a new append vec at the same slot OR there is no account index entry for this account in the append vec we are shrinking. These accounts should already have been removed.
So, it is inefficient and unnecessary to go over all accounts in the append vec we are shrinking that are alive and work on reclaiming the space in the append vec. Because we are throwing away the entire append vec already.

**Review by HaoranYi — APPROVED** [review]
lgtm.

**Pull request merged**